newIDE/app/src/AssetStore/ExtensionStore/ExtensionInstallDialog.js

```diff
@@ -55,7 +58,13 @@ const ExtensionInstallDialog = ({
   onInstall,
   onEdit,
   alreadyInstalled,
+  project,
 }: Props) => {
+  const installedExtensions = new Set(
+    enumerateEventsFunctionsExtensions(project).map(ext => ext.getName())
+  );
```

Comment:
Could you useMemo to avoid recomputing these updates/installed extensions at each render?
Could we also return the installedExtensions from the hook so we avoid to enumerate twice the extensions?

Reply:
I hadn't done it because the project is not something we can use as a dependency for react hooks or that will trigger a rerender on an update of the list of extensions, so the memo could keep a wrong outdated memoized list of installed extensions. While it is very unlikely for something else to happen while in this menu, and not a huge issue to have that button mislabeled, it is dangerous to make these kinds of assumptions that "this will never happen here" imo. Especially if one day we add remote collaboration and a lot of stuff could be done with the project while someone does something else on a completely different menu.

Reply:
There are a few things that would allow to avoid a future problem of outdated memoized list:
- Any change could generate a new JS object (in the sense, not the same reference) so that any useMemo/useContext is updated.
- Maybe we'll have a solution like `const project = useCollaborativeProject()` so that any dialog will progressively be made collaborative.
- In any case, the issue of "this project is potentially outdated" is limited to this dialog - you just need to close the dialog and re-open it to avoid the issue.

Not a big deal here anyway, but some dialogs are lagging so I'm now more carful about this.
Note that when we add collaborative updates of a `project`, useMemo(..., [project]) should *still* work. It's not because you have a project that can be updated at any time that you don't want to optimise your renderings (in which case you probably want to use what I said before).